docroot/modules/custom/uiowa_apr/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,230 @@
+      '#required' => FALSE,
+    ];
+
+    $form['directory_show_switcher'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Show List Switcher'),
+      '#return_value' => TRUE,
+      '#default_value' => $this->config('uiowa_apr.settings')->get('directory.show_switcher') ?? FALSE,
+      '#description' => $this->t('Flag to show or hide the control that allows the user to switch between list views.'),
+    ];
+
+    $form['publications_path'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Publications Path'),
+      '#default_value' => $this->config('uiowa_apr.settings')->get('publications.path') ?? '/apr/publications',
+      '#description' => $this->t("Path for the site's APR publications directory."),
+      '#required' => TRUE,
+    ];
+
+    $form['publications_title'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Publications Title'),
+      '#default_value' => $this->config('uiowa_apr.settings')->get('publications.title') ?? 'Research',
+      '#description' => $this->t("Page title for the site's primary publications directory."),
+      '#required' => TRUE,
+    ];
+
+    $form['publications_page_size'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Publications Page Size'),
+      '#default_value' => $this->config('uiowa_apr.settings')->get('publications.page_size') ?? 10,
+      '#min' => 5,
+      '#max' => 50,
+      '#description' => $this->t('Number of entries per page of the publications directory. Min: 5, Max: 50.'),
+      '#required' => TRUE,
+    ];
+
+    $form['publications_departments'] = [
+      '#type' => 'textarea',
+      '#rows' => '5',
+      '#cols' => '100',
+      '#title' => $this->t('Publications Departments'),
+      '#default_value' => $this->config('uiowa_apr.settings')->get('publications.departments') ?? '',
+      '#description' => $this->t('Customize the list of departments exposed by the publications tool. Expects a JSON array. Value attributes in the array must match a department name in APR. Use the text attribute to customize the text the user will see.'),
+      '#attributes' => ['placeholder' => "[{text: 'Economics', value: 'Economics'}]"],
+      '#required' => FALSE,
+    ];
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    $fields = [
+      'directory_path',
+      'publications_path',
+    ];
+
+    foreach ($fields as $field) {
+      $path = $this->aliasCleaner->cleanAlias($form_state->getValue($field));
+
+      /** @var \Drupal\Core\Url $url */
+      $url = $this->pathValidator->getUrlIfValid($path);
+
+      // If $url is anything besides FALSE then the path is already in use.
+      if ($url && substr($url->getRouteName(), 0, 9) != 'uiowa_apr') {
```

Comment:
@joewhitsitt Initially, I copied the path validation from sitenow_people et al. but it did not catch route conflicts with those modules. For example, changing the APR path to people did not catch the conflict with sitenow_people default path. Curious what you think of this method and also if you think we should add any specific help text related to sitenow_people.

Reply:
if this works to catch view pages, etc. go with it. i remember you bringing this up previously with sitenow_people and sitenow_articles but we never revisited it. what kind of help text did you have in mind?

Reply:
Not sure if we wanted to highlight any confusion with sitenow_people configuration or not. I guess we can probably add that to the docs.